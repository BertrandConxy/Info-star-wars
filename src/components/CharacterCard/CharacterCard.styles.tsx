import styled from 'styled-components'
import { theme } from '../../infrastructure/theme'
import { Link } from 'react-router-dom'

export const CharacterCardStyled = styled.div`
  border: 1px solid white;
  margin: ${theme.space[2]};
  padding: ${theme.space[2]};
  padding-left: ${theme.space[3]};
  @media screen and (max-width: 768px) {
    padding-left: ${theme.space[2]};
  }
`
export const Title = styled.h2`
  margin-bottom: ${theme.space[2]};
`

export const Content = styled.p`
  font-weight: bold;
`

export const Span = styled.span`
  font-weight: normal;
`

export const LinkContainer = styled(Link)`
  color: ${theme.colors.brand.primary};
`
