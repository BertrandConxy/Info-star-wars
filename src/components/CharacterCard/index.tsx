import React from 'react'

import {
  CharacterCardStyled,
  Title,
  Content,
  Span,
  LinkContainer,
} from './CharacterCard.styles'
import { iProps } from '../../typeDefs/characterCardProps'

export default function CharacterCard(props: iProps) {
  const { id, name, birthYear, hairColor, gender, eyeColor } = props
  return (
    <LinkContainer to={`/characters/${id}`}>
      <CharacterCardStyled>
        <Title>{name}</Title>
        <Content>
          Birth Year: <Span>{birthYear}</Span>
        </Content>
        <Content>
          Eye Color: <Span>{eyeColor}</Span>
        </Content>
        <Content>
          Hair Color: <Span>{hairColor}</Span>
        </Content>
        <Content>
          Gender: <Span>{gender}</Span>
        </Content>
      </CharacterCardStyled>
    </LinkContainer>
  )
}
