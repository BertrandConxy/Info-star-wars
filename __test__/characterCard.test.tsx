import React from 'react'
import { render, screen } from '@testing-library/react'
import CharacterCard from '../src/components/CharacterCard/index'
import { MemoryRouter } from 'react-router-dom'

const mockProps = {
  id: 1,
  name: 'Luke Skywalker',
  birthYear: '19 BBY',
  hairColor: 'Blond',
  gender: 'Male',
  eyeColor: 'Blue',
}

describe('CharacterCard', () => {
  beforeEach(() => {
    render(
      <MemoryRouter>
        <CharacterCard {...mockProps} />
      </MemoryRouter>,
    )
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterCard {...mockProps} />
      </MemoryRouter>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the character name', () => {
    expect(screen.getByText('Luke Skywalker')).toBeInTheDocument()
  })

  it('renders the birth year', () => {
    expect(screen.getByText('Birth Year:')).toBeInTheDocument()
    expect(screen.getByText('19 BBY')).toBeInTheDocument()
  })

  it('renders the eye color', () => {
    expect(screen.getByText('Eye Color:')).toBeInTheDocument()
    expect(screen.getByText('Blue')).toBeInTheDocument()
  })

  it('renders the hair color', () => {
    expect(screen.getByText('Hair Color:')).toBeInTheDocument()
    expect(screen.getByText('Blond')).toBeInTheDocument()
  })

  it('renders the gender', () => {
    expect(screen.getByText('Gender:')).toBeInTheDocument()
    expect(screen.getByText('Male')).toBeInTheDocument()
  })

  it('renders a link to the character details page', () => {
    expect(screen.getByRole('link')).toHaveAttribute('href', '/characters/1')
  })
})
