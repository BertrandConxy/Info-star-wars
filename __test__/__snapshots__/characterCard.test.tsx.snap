// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="sc-eDDNvR eKNcDH"
    href="/characters/1"
  >
    <div
      class="sc-beqWaB jDRaRx"
    >
      <h2
        class="sc-gueYoa bzjvzv"
      >
        Luke Skywalker
      </h2>
      <p
        class="sc-dmqHEX cbNNmG"
      >
        Birth Year: 
        <span
          class="sc-hLseeU izxruz"
        >
          19 BBY
        </span>
      </p>
      <p
        class="sc-dmqHEX cbNNmG"
      >
        Eye Color: 
        <span
          class="sc-hLseeU izxruz"
        >
          Blue
        </span>
      </p>
      <p
        class="sc-dmqHEX cbNNmG"
      >
        Hair Color: 
        <span
          class="sc-hLseeU izxruz"
        >
          Blond
        </span>
      </p>
      <p
        class="sc-dmqHEX cbNNmG"
      >
        Gender: 
        <span
          class="sc-hLseeU izxruz"
        >
          Male
        </span>
      </p>
    </div>
  </a>
</DocumentFragment>
`;
